### PR TITLE
Add support for boto3

### PR DESCRIPTION
**Pull request by Jamim**
Hello @yurishkuro,

It would be nice to have support for [boto3](https://github.com/boto/boto3) in the upstream.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,12 +17,19 @@ install:
   - make install-ci
 
 services:
+  - docker
   - postgresql
   - redis
 
 before_script:
+  - docker run -d -p 8000:8000 dwmkerr/dynamodb
```
Comment (yurishkuro) [inline .travis.yml]: official image would be betterr: https://hub.docker.com/r/amazon/dynamodb-local/

File: .travis.yml [hunk 2]
```diff
@@ -17,12 +17,19 @@ install:
...
+  - sleep 1  # let DynamoDB start
```
Comment (yurishkuro) [inline .travis.yml]: speed in travis is unpredictable, there's no guarantee 1s would be enough.

can we instead use some simple health check script?

File: README.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ The module name is `opentracing_instrumentation`.
 ### Supported client frameworks
 
 The following libraries are instrumented for tracing in this module:
+ * `boto3`
```
Comment (yurishkuro) [inline README.md]: let's add a small comment explaining what it is (the other module names are more obvious), and maybe turn it into a link

File: .travis.yml (hunk 1 above)
Comment (Jamim) [reply .travis.yml]: Thank you for the suggestion!
I've updated the PR.

File: .travis.yml (hunk 2 above)
Comment (Jamim) [reply .travis.yml]: I've added a simple health check.

File: README.md (hunk 3 above)
Comment (Jamim) [reply README.md]: I've turned `boto3` into a link and added a short description.

I can also create an extra PR that will do the same for the rest of the list.
What do you think?

**Comment by Jamim** [comment]
> is it necessary to support this fairly vendor-specific lib (boto3) here instead of as in a standalone module?

Yes, I believe that `opentracing_instrumentation` library is the right place for it.

`opentracing_instrumentation` provides an easy way to enable tracing for a bunch of third party libraries. I suppose that it will be much more complicated to maintain and also to use a standalone module to support `boto3`.

> What are the benefits of having it here?

`boto3` is [a quite popular library](https://github.com/boto/boto3/network/dependents) that is [widely used](https://pypistats.org/packages/boto3), so supporting it will make `opentracing_instrumentation` more useful for many users.

**Comment by yurishkuro** [comment]
Fair enough. Can you make the change to use docker-compose for DBs instead of Travis services?

**Comment by Jamim** [comment]
> Can you make the change to use docker-compose for DBs instead of Travis services?

I'm working on it currently.
But I want to keep `postgres` and `redis` as Travis services because I'd like to avoid adding a few more health checks for them.
What do you think?

**Comment by Jamim** [comment]
Hi @yurishkuro,

`docker-compose.yml` is ready for the review.

**Comment by Jamim** [comment]
ping @yurishkuro

**Comment by Jamim** [comment]
Hi @yurishkuro,

Are any improvements required?

**Pull request merged**